### PR TITLE
[lgwebos] Avoid thing remainiong in INITIALIZING state at startup whe…

### PR DESCRIPTION
**Pull request by lolodomo**
…n TV is off

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
This PR fixed a problem that the thing was remaining in INITIALIZING state after the binding was started until the TV was switched on.

File: bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSTVSocket.java [hunk 1]
```diff
@@ -97,7 +97,8 @@
         REGISTERED
     }
 
-    private State state = State.DISCONNECTED;
+    @Nullable
+    private State state = null;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSTVSocket.java]:
```suggestion
    private @Nullable State state = null;
```

**Comment by sprehn** [comment]
Hi,
I disagree with making state nullable. especially when 'null' means 'DISCONNECTED' 

```
public State getState() {
         State state = this.state;
         return state == null ? State.DISCONNECTED : state;
}
```

**Comment by lolodomo** [comment]
The problem was that if the state is initialized with DISCONNECTED, the first call to setState (when timeout) with DISCONNECTED does not trigger any change on the thing handler because the value is unchanged and the thing status is not updated.
This control of change of value has a sense, except the first time.

**Comment by lolodomo** [comment]
If you don't like null, we can define a new state value like UNSET and set it as initial value.
Is it ok for you ?

**Comment by sprehn** [comment]
Hi, was just thinking through the options. Initially had the same idea of adding a dedicated INIT state.

but what about a simple 
updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "TV is off");
at the end of LGWebOSHandler:initialize()

**Comment by lolodomo** [comment]
Yes, that is a solution, I even thought about it before deciding to implement the solution in this PR.
The only disadvantage of your solution (updateStatus in method initialize) is that the thing status will first be set to OFFLINE and then ONLINE even when the the TV is on when the binding starts.
My solution was avoiding that.
But ok we can go with your solution which is simpler, just one line of code to add.

**Comment by lolodomo** [comment]
I see that you already added the updateStatus in your other PR, so I close this PR.

**Pull request closed without merging**